CoreAnalyticSchema/src/au/gov/asd/tac/constellation/graph/schema/analytic/compatibility/AnalyticSchemaV5UpdateProvider.java

```diff
@@ -0,0 +1,87 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.graph.schema.analytic.compatibility;
+
+import au.gov.asd.tac.constellation.graph.GraphElementType;
+import au.gov.asd.tac.constellation.graph.StoreGraph;
+import au.gov.asd.tac.constellation.graph.schema.SchemaFactory;
+import au.gov.asd.tac.constellation.graph.schema.SchemaFactoryUtilities;
+import au.gov.asd.tac.constellation.graph.schema.analytic.AnalyticSchemaFactory;
+import au.gov.asd.tac.constellation.graph.schema.analytic.concept.ClusteringConcept;
+import au.gov.asd.tac.constellation.graph.versioning.SchemaUpdateProvider;
+import au.gov.asd.tac.constellation.graph.versioning.UpdateProvider;
+import org.openide.util.lookup.ServiceProvider;
+
+/**
+ * Update provider to ensure that previous attributes using the spelling "colour" are updated
+ * to the new spelling "color" to be consistent throughout the application
+ *
+ * @author Delphinus8821
+ */
+@ServiceProvider(service = UpdateProvider.class)
+public class AnalyticSchemaV5UpdateProvider extends SchemaUpdateProvider {
+
+    public static final int SCHEMA_VERSION_THIS_UPDATE = 5;
+
+    private static final String K_TRUSS_COLOUR = "Cluster.KTruss.Colour";
+    private static final String HIERARCHICAL_COLOUR = "Cluster.Hierarchical.Colour";
+    private static final String INFOMAP_COLOUR = "Cluster.Infomap.Colour";
+
+    @Override
+    protected SchemaFactory getSchema() {
+        return SchemaFactoryUtilities.getSchemaFactory(AnalyticSchemaFactory.ANALYTIC_SCHEMA_ID);
+    }
+
+    @Override
+    public int getFromVersionNumber() {
+        return AnalyticSchemaV4UpdateProvider.SCHEMA_VERSION_THIS_UPDATE;
+    }
+
+    @Override
+    public int getToVersionNumber() {
+        return SCHEMA_VERSION_THIS_UPDATE;
+    }
+
+    @Override
+    protected void schemaUpdate(final StoreGraph graph) {
+
+        // Retrieve the attributes IDs of the previous attributes
+        final int oldKTrussColourAttributeId = graph.getAttribute(GraphElementType.VERTEX, K_TRUSS_COLOUR);
+        final int oldHierarchicalColourAttributeId = graph.getAttribute(GraphElementType.VERTEX, HIERARCHICAL_COLOUR);
+        final int oldInformapColourAttributeId = graph.getAttribute(GraphElementType.VERTEX, INFOMAP_COLOUR);
+
+        // retrieve the new values for the graph_labels_bottom and set it to the new values
+        final int newKTrussColorAttributeId = ClusteringConcept.VertexAttribute.K_TRUSS_COLOR.ensure(graph);
+        final String kTrussColorValue = graph.getStringValue(oldKTrussColourAttributeId, 0);
+        graph.setStringValue(newKTrussColorAttributeId, 0, kTrussColorValue);
```

Comment:
A couple of things here:
1) ensure will create the attribute if it doesn't exist but we only want to edit it if it does already exist. Are you able to change the ensure to a get and then only edit if the attribute is found
2) id 0 is the graph and so for vertex and transaction attributes, you'll get an array out of bounds exception for the get and set value. Are you able to iterate through the relevant graph elements to set each graph element appropriately? You can refer to some of the previous update providers for doing that.

This will be the same for the other attributes in this update provider